pkg/lokifrontend/frontend/v2/frontendv2pb/frontend.proto

```diff
@@ -17,7 +18,10 @@ service FrontendForQuerier {
 
 message QueryResultRequest {
   uint64 queryID = 1;
-  httpgrpc.HTTPResponse httpResponse = 2;
+  oneof response {
+    httpgrpc.HTTPResponse httpResponse = 2 [deprecated = true]; // or mark as reserved
```

Comment:
What impact does the `deprecated` flag here have?

Reply:
I think it just results in a build warning? From the protodocs:
> deprecated (field option): If set to true, indicates that the field is deprecated and should not be used by new code. In most languages this has no actual effect. In Java, this becomes a @Deprecated annotation. For C++, clang-tidy will generate warnings whenever deprecated fields are used. In the future, other language-specific code generators may generate deprecation annotations on the field’s accessors, which will in turn cause a warning to be emitted when compiling code which attempts to use the field. If the field is not used by anyone and you want to prevent new users from using it, consider replacing the field declaration with a [reserved](https://protobuf.dev/programming-guides/proto3/#fieldreserved) statement.

https://protobuf.dev/programming-guides/proto3/#options and then scroll a bit, bullet list not an addressable md header unfortunately

Reply:
The linter will also fail. I can leave it out for now but do think we should start using the "pure" protobuf.

Reply:
I'm not sure what "pure" protobuf means here. Replacing gogoproto?